File: nexgen_msp/utils/device_status.py
"""What each status of a machine means, written once for everything that asks."""

# out in the field, in somebody's hands
DEPLOYED_STATUSES = ("Active",)

# on the shelf at the customer, free to be given to somebody
AVAILABLE_STATUSES = ("Pending", "Stock")

# not in service: nobody holds one of these
UNAVAILABLE_STATUSES = ("Returned", "Damaged", "Retired", "Lost")

# the end of the line
TERMINAL_STATUSES = ("Retired",)

RETIRABLE_FROM = ("Active", "Stock", "Damaged", "Lost")
REINSTATABLE_FROM = ("Retired", "Damaged", "Lost", "Returned")


def is_deployed(status):
	return status in DEPLOYED_STATUSES


def is_out_of_service(status):
	return status in UNAVAILABLE_STATUSES

File: frontend/src/features/internal/utils/deviceStatus.ts
export const DEPLOYED_STATUSES = ['Active'] as const;
export const AVAILABLE_STATUSES = ['Pending', 'Stock'] as const;
export const UNAVAILABLE_STATUSES = ['Returned', 'Damaged', 'Retired', 'Lost'] as const;
export const RETIRABLE_FROM = ['Active', 'Stock', 'Damaged', 'Lost'] as const;
export const REINSTATABLE_FROM = ['Retired', 'Damaged', 'Lost', 'Returned'] as const;

export const isDeployed = (status: string) => (DEPLOYED_STATUSES as readonly string[]).includes(status);
export const isAvailable = (status: string) => (AVAILABLE_STATUSES as readonly string[]).includes(status);
export const isOutOfService = (status: string) => (UNAVAILABLE_STATUSES as readonly string[]).includes(status);
export const canRetire = (status: string) => (RETIRABLE_FROM as readonly string[]).includes(status);
export const canReinstate = (status: string) => (REINSTATABLE_FROM as readonly string[]).includes(status);
